core/planner.c

```diff
@@ -1017,32 +1015,36 @@ bool plan(struct diveplan *diveplan, struct dive *dive, int timestep, struct dec
 							laststoptime = 12 * 60;
 							new_clock = clock + laststoptime;
 							o2breaking = true;
-							breaktime = 0;
+							o2break_next = true;
+							o2break_done = true;
 							breakfrom_cylinder = current_cylinder;
 							if (is_final_plan)
 								plan_add_segment(diveplan, laststoptime, depth, current_cylinder, po2, false);
 							previous_point_time = clock + laststoptime;
 							current_cylinder = break_cylinder;
 							gas = dive->cylinder[current_cylinder].gasmix;
+						} else if (!o2break_done && laststoptime < prefs.min_switch_duration) {
+							/* Stop for the minimum duration to switch gas now because we didn't do o2breaks */
+							add_segment(depth_to_bar(depth, dive),
+								&dive->cylinder[current_cylinder].gasmix,
+								prefs.min_switch_duration - laststoptime, po2, dive, prefs.decosac);
+							clock += prefs.min_switch_duration - laststoptime;
 						}
-					} else {
-						if (breaktime >= 0) {
-							if (laststoptime >= 6 * 60) {
-								laststoptime = 6 * 60;
-								new_clock = clock + laststoptime;
-								o2breaking  = true;
-								o2time = 0;
-								if (is_final_plan)
-									plan_add_segment(diveplan, laststoptime, depth, current_cylinder, po2, false);
-								previous_point_time = clock + laststoptime;
-								current_cylinder = breakfrom_cylinder;
-								gas = dive->cylinder[current_cylinder].gasmix;
-								breaktime = -1;
-							}
+					} else if (o2break_next) {
```

Comment:
Coding style: Please write this as
} else {
    if (...

Reply:
I don't agree here.

I think you rather should lift up the inner if statement into this instead.

I also think there is too much inline code here, so if its possible, we should look at lifting those out to separate helpers to keep the general code flow easier on the eyes.

Reply:
@atdotde - no, we do allow `} else if (...) {`